src/core/mac/mac.cpp

```diff
@@ -1303,6 +1303,13 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
         otDumpDebgMac("TX", aFrame.GetHeader(), aFrame.GetLength());
         FinishOperation();
         Get<MeshForwarder>().HandleSentFrame(aFrame, aError);
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2
+        if (aError == OT_ERROR_NONE && Get<Mle::Mle>().GetParent().IsEnhancedKeepAliveSupported() &&
+            aFrame.GetSecurityEnabled() && aAckFrame != NULL)
```

Comment:
Should check `aFrame.otMacFrameIsData()`?

Reply:
@brianljt yes, otherwise we would be sending MAC Data Requests one after each other.

Reply:
I think the operation is transmitting data only.

Reply:
I agree with @bukepo . Data polls acks are handled under `case kOperationTransmitPoll` above. Here we are under   `case kOperationTransmitDataDirect` (which means this is a recieved ack for a data frame tx and not a data poll tx).